xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererMediaCodecSurface.cpp

```diff
@@ -63,6 +63,18 @@ bool CRendererMediaCodecSurface::Configure(const VideoPicture &picture, float fp
   CalculateFrameAspectRatio(picture.iDisplayWidth, picture.iDisplayHeight);
   SetViewMode(m_videoSettings.m_ViewMode);
 
+  // Configure GUI/OSD for HDR PQ when display is in HDR PQ mode
+  if (picture.color_transfer == AVCOL_TRC_SMPTE2084)
+  {
+    if (CServiceBroker::GetWinSystem()->IsHDRDisplay())
+      CServiceBroker::GetWinSystem()->GetGfxContext().SetTransferPQ(true);
+  }
+  else if (picture.hdrType == StreamHdrType::HDR_TYPE_DOLBYVISION)
+  {
+    if (CServiceBroker::GetWinSystem()->GetDisplayHDRCapabilities().SupportsDolbyVision())
+      CServiceBroker::GetWinSystem()->GetGfxContext().SetTransferPQ(true);
+  }
```

Comment:
Also is there a similar code path to modify for software decoding?

Reply:
In Android there are not much options: Android devices are not powerful enough to do software HDR video decoding / rendering. If no HDR or DV display available then devices _may_ do HDR to SDR mapping (HW based) but this is already taken into account as then `SetTransferPQ(true)` is not reached.

GUI Setting (old name) "Use HDR display capabilities" is half broken in Android because when MediaCodecSurface is used, output is always "HDR passthrough" if system has HDR display (even with setting disabled).

Intentionally only is handled RenderMediCodecSurface code path because SWR path anyway is not usable for HDR on Android devices.

Reply:
Are there PVR type of situations where a stream could change back and forth between HDR and SDR? If so, need an `else SetTransferPQ(false);` or similar.

I was wondering about dovi fallback to hdr10 but that seems covered already, right ?(streams that have HDR10 compatibility)

HLG streams likely need the luminance adjustment too.

Reply:
> Are there PVR type of situations where a stream could change back and forth between HDR and SDR? If so, need an else SetTransferPQ(false); or similar.

`CRendererMediaCodecSurface::Configure` only is called one time when opening a stream.

> HLG streams likely need the luminance adjustment too. 

Not in Android. This was already tested. In Windows is need because not supports HLG output and is converted to HDR10 (when display supports HDR). In Android, GUI has correct bright, while playing HLG streams before and after this PR.